google/cloud/bigquery/_helpers.py

```diff
@@ -310,17 +312,13 @@ def _json_from_json(value, field):
 
 
 def _range_element_from_json(value, field):
-    """Coerce 'value' to a range element value, if set or not nullable."""
+    """Coerce 'value' to a range element value."""
     if value == "UNBOUNDED":
         return None
-    elif field.element_type == "DATE":
-        return _date_from_json(value, None)
-    elif field.element_type == "DATETIME":
-        return _datetime_from_json(value, None)
-    elif field.element_type == "TIMESTAMP":
-        return _timestamp_from_json(value, None)
+    if field.element_type in _SUPPORTED_RANGE_ELEMENTS:
+        return _CELLDATA_FROM_JSON[field.element_type](value, field.element_type)
     else:
-        raise ValueError(f"Unsupported range field type: {value}")
+        raise ValueError(f"Unsupported range field type: {field.element_type}")
```

Comment:
nit: should we change this to indicate the element type is unsupported, rather than "field type"?

Reply:
Indeed, I'll change it to be consistent with the name of the field.